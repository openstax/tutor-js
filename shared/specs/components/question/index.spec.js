import Factories from '../../factories';
import Snapshot from 'react-test-renderer';
import Question from '../../../src/components/question';

// eslint-disable-next-line react/prop-types
jest.mock('../../../src/components/html', () => ({ className, html }) =>
    html ? <div className={className} dangerouslySetInnerHTML={{ __html: html }} /> : null
);

describe('Question Component', function() {
    let props = null;

    beforeEach(() => {
        const exercise = Factories.exercise();
        props = {
            question: exercise.questions[0],
            onChange: jest.fn(),
            type: 'student',
        };
    });

    it('renders and matches snapshot', () => {
        props.type = 'teacher-preview'; // hack to hide instructions
        expect(Snapshot.create(<Question {...props} />).toJSON()).toMatchSnapshot();
    });

    it('highlights when answer is clicked', () => {
        const q = mount(<Question {...props} />);
        const answer = q.find('.answers-answer').at(1);
        expect(answer.hasClass('answer-checked')).toEqual(false);
        answer.find('input').simulate('change', { target: { checked: true } });
        expect(props.onChange).toHaveBeenCalled();
    });

    it('renders the context when given', () => {
        props.context = 'THIS IS SOME STUFF TO EXPLAIN';
        const q = mount(<Question {...props} />);
        expect(q.text()).toContain('EXPLAIN');
    });
});
