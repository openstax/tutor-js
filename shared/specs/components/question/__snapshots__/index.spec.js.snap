// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Component renders and matches snapshot 1`] = `
<div
  className="openstax-question"
>
  <div
    className="question-stem"
    dangerouslySetInnerHTML={
      Object {
        "__html": "What conditions imply that we can use classical physics without considering special relativity or quantum mechanics?",
      }
    }
  />
  <div
    className="answers-table"
  >
    <div
      className="openstax-answer"
    >
      <section
        className="answers-answer disabled"
        role="region"
      >
        <div
          className="correct-incorrect"
        />
        <label
          className="answer-label"
          htmlFor="1-option-0"
          onKeyPress={[Function]}
        >
          <button
            aria-label="Choice a:"
            className="answer-letter"
            data-test-id="answer-choice-a"
            disabled={true}
          >
            a
          </button>
          <div
            className="answer-answer"
          >
            <div
              className="answer-content"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<ol>
  <li>matter is moving at speeds of less than roughly 1% the speed of light,</li>
  <li>objects are too small to be seen with the naked eye, and</li>
  <li>there is the involvement of only a weak gravitational field</li>
</ol>
",
                }
              }
            />
          </div>
        </label>
      </section>
    </div>
    <div
      className="openstax-answer"
    >
      <section
        className="answers-answer disabled"
        role="region"
      >
        <div
          className="correct-incorrect"
        />
        <label
          className="answer-label"
          htmlFor="1-option-1"
          onKeyPress={[Function]}
        >
          <button
            aria-label="Choice b:"
            className="answer-letter"
            data-test-id="answer-choice-b"
            disabled={true}
          >
            b
          </button>
          <div
            className="answer-answer"
          >
            <div
              className="answer-content"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<ol>
  <li>matter is moving at speeds greater than roughly 1% the speed of light,</li>
  <li>objects are large enough to be seen with the naked eye, and</li>
  <li>there is the involvement of a strong gravitational field</li>
</ol>
",
                }
              }
            />
          </div>
        </label>
      </section>
    </div>
    <div
      className="openstax-answer"
    >
      <section
        className="answers-answer disabled"
        role="region"
      >
        <div
          className="correct-incorrect"
        />
        <label
          className="answer-label"
          htmlFor="1-option-2"
          onKeyPress={[Function]}
        >
          <button
            aria-label="Choice c:"
            className="answer-letter"
            data-test-id="answer-choice-c"
            disabled={true}
          >
            c
          </button>
          <div
            className="answer-answer"
          >
            <div
              className="answer-content"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<ol>
  <li>matter is moving at speeds of less than roughly 1% the speed of light,</li>
  <li>objects are large enough to be seen with the naked eye, and</li>
  <li>there is the involvement of a strong gravitational field</li>
</ol>
",
                }
              }
            />
          </div>
        </label>
      </section>
    </div>
  </div>
  <div
    className="detailed-solution"
  >
    <div
      className="header"
    >
      Detailed solution
    </div>
    <div
      className="solution"
      dangerouslySetInnerHTML={
        Object {
          "__html": "four",
        }
      }
    />
  </div>
</div>
`;
